### PR TITLE
Speed up Predict.process_*()

### PR DESCRIPTION
**Pull request by frankenjoe**
Relates to #100 

Speeds up `Predict.process_*()` by avoiding calls to `pd.concat()` on a lot of small `pd.Series` objects. Instead, we now concatenate lists and create the final `pd.Series` from those.

### Benchmark

```python
import time
import audb
import audinterface


db = audb.load(
    'emodb',
    version='1.3.0',
    format='wav',
    sampling_rate=16000,
    mixdown=True,
)

def process_func(x, sr):
    return [x.mean(), x.std()]

process = audinterface.Process(process_func=process_func)

t = time.time()
process.process_files(db.files)
print(time.time() - t)
```

* main branch: ~4.5 s
* this branch: ~0.3 s

### CODE REVIEW

File: audinterface/core/process.py [hunk 1]
```diff
@@ -411,7 +415,16 @@ def process_files(
             task_description=f'Process {len(files)} files',
         )
         self.verbose = verbose
-        return pd.concat(y)
+
+        y = list(itertools.chain.from_iterable([x[0] for x in xs]))
```
Comment (hagenw) [inline audinterface/core/process.py:419]: Why is 

```python
y = [x[0] for x in xs]
```
not working?

Comment (frankenjoe) [reply audinterface/core/process.py:419]: We have a list of lists that we want to flatten.

Comment (hagenw) [reply audinterface/core/process.py:419]: Ah, ok, so `x` is not a tuple containing single `(y, file, start, end)` entries, but `(y, files, starts, ends)` lists, correct?

Comment (frankenjoe) [reply audinterface/core/process.py:419]: yes

**Comment by hagenw** [comment]
We have now code that we repeat at some places. In principle we could replace it by the following function:

```python
def _create_series_from_list(xs, skip_file_level=False):

    y = list(itertools.chain.from_iterable([x[0] for x in xs]))
    starts = list(itertools.chain.from_iterable([x[2] for x in xs]))
    ends = list(itertools.chain.from_iterable([x[3] for x in xs]))

    if skip_file_level:
        index = utils.signal_index(starts, ends)
    else:
        files = list(itertools.chain.from_iterable([x[1] for x in xs]))
        index = audformat.segmented_index(files, starts, ends)

    return pd.Series(y, index)
```

But I'm not completely sure if this might it harder to understand what is going on inside specific methods, e.g.

```python
        xs = audeer.run_tasks(
            self._process_file,
            params,
            num_workers=self.num_workers,
            multiprocessing=self.multiprocessing,
            progress_bar=self.verbose,
            task_description=f'Process {len(index)} segments',
        )
        y = _create_series_from_list(xs)
```

Did you already thought about this?

**Comment by frankenjoe** [comment]
Yes, I thought about it, but as you say, it makes the code more complicated to follow and it's very little code we duplicate. So I think I would leave it as is.

**Pull request merged**